xbmc/music/MusicDatabase.cpp

```diff
@@ -3282,6 +3433,67 @@ bool CMusicDatabase::GetYearsNav(const std::string& strBaseDir, CFileItemList& i
   return false;
 }
 
+bool CMusicDatabase::GetRolesNav(const std::string& strBaseDir, CFileItemList& items, const Filter &filter /* = Filter() */)
+{
+  try
+  {
+    if (NULL == m_pDB.get()) return false;
+    if (NULL == m_pDS.get()) return false;
+
+    Filter extFilter = filter;
+    CMusicDbUrl musicUrl;
+    SortDescription sorting;
+    if (!musicUrl.FromString(strBaseDir) || !GetFilter(musicUrl, extFilter, sorting))
+      return false;
+
+    // get roles with artists having that role
+    std::string strSQL = "SELECT idRole, strRole FROM Role WHERE EXISTS "
+                         "(SELECT 1 FROM song_artist WHERE song_artist.idRole = Role.idRole)";
+
+    if (!BuildSQL(strSQL, extFilter, strSQL))
+      return false;
+
+    // run query
+    CLog::Log(LOGDEBUG, "%s query: %s", __FUNCTION__, strSQL.c_str());
+    if (!m_pDS->query(strSQL))
+      return false;
+    int iRowsFound = m_pDS->num_rows();
+    if (iRowsFound == 0)
+    {
+      m_pDS->close();
+      return true;
+    }
+
+    // get data from returned rows
+    while (!m_pDS->eof())
+    {
+      std::string labelValue = m_pDS->fv(1).get_asString();
+      CFileItemPtr pItem(new CFileItem(labelValue));
```

Comment:
Unfortunately make_shared doesn't have a version for string argument, or have I misunderstood?